metapool/sample_sheet.py

```diff
@@ -1241,30 +1232,88 @@ def _table_contains_katharoseq_samples(self, table):
         # this helper method will return True only if a katharo-control
         # sample is found. Note criteria for this method should be kept
         # consistent w/the above method (contains_katharoseq_samples).
-        return table[_SS_SAMPLE_NAME_KEY].str.startswith('katharo').any()
+        is_katharos = table[_SS_SAMPLE_NAME_KEY].apply(self._is_katharo_name)
+        return is_katharos.any()
 
-    def _get_expected_columns(self, table=None):
+    def _get_expected_data_columns(self, table=None):
         if table is None:
             # if [Data] section contains katharoseq samples, add the expected
             # additional katharoseq columns to the official list of expected
             # columns before validation or other processing begins.
             if self.contains_katharoseq_samples():
-                return self.data_columns + self.optional_katharoseq_columns
+                return self._data_columns + self._optional_katharoseq_columns
+
         else:
             # assume that there are no samples added to this object yet. This
             # means that self.contains_katharoseq_samples() will always return
             # False. Assume table contains a list of samples that may or may
             # not contain katharoseq controls.
             if self._table_contains_katharoseq_samples(table):
-                return self.data_columns + self.optional_katharoseq_columns
+                return self._data_columns + self._optional_katharoseq_columns
+
+        return self._data_columns
+
 
-        return self.data_columns
+class AmpliconSampleSheet(KLSampleSheet):
+    _HEADER = {
+        'IEMFileVersion': '4',
+        _SHEET_TYPE_KEY: _DUMMY_SHEET_TYPE,
+        _SHEET_VERSION_KEY: '0',
+        # Per MacKenzie's request, these are removed if found.
+        # 'Investigator Name': 'Knight',
+        # 'Experiment Name': 'RKL_experiment',
+        'Date': None,
+        'Workflow': 'GenerateFASTQ',
+        'Application': 'FASTQ Only',
+        _ASSAY_KEY: _AMPLICON,
+        'Description': '',
+        'Chemistry': 'Default',
+    }
+
+    _CARRIED_PREP_COLUMNS = (EXPT_DESIGN_DESC_KEY, 'i5_index_id',
+                             'i7_index_id', 'index', 'index2',
+                             'library_construction_protocol',
+                             SAMPLE_NAME_KEY,
+                             'sample_plate', 'sample_project',
+                             'well_description', 'Sample_Well')
+
+    def __init__(self, path=None):
+        super().__init__(path)
+        self._remapper = {
+            'sample sheet Sample_ID': SS_SAMPLE_ID_KEY,
+            'Sample': _SS_SAMPLE_NAME_KEY,
+            PM_PROJECT_PLATE_KEY: 'Sample_Plate',
+            'Well': 'Sample_Well',
+            'Name': 'I7_Index_ID',
+            'Golay Barcode': 'index',
+            PM_PROJECT_NAME_KEY: _SS_SAMPLE_PROJECT_KEY,
+        }
+
+
+class MetagenomicSampleSheetv102(
```

Comment:
Based on this definition, do we need to have MetagenomicSampleSheetv101 and MetagenomicSampleSheetv102? Because I don't believe we're run any v101 sheets through production, and the definitions below define the only difference between the two as included support for Katharoseq controls. Since katharoseq controls are optional then all v101 sheets are valid v102 sheets as well, aren't they?

Reply:
Hahahaha, @charles-cowart, great minds think alike!  Based on the same line of thinking, I had actually killed off v102, changed v101 to derive from KLSampleSheetWithContext, and switched over all the tests and everything (which went swimmingly) ... but I then discovered that trying to run the matrix tube **notebook** to create a v101 samplesheet *with* katharoseq controls (instead of without them) caused an error due to not having all the required katharoseq columns (those `katharo_description` and `platemap_generation_date` columns that the group has been discussing on slack).  Since I couldn't get an immediate answer to what the columns should be and how they should be generated, and I needed to get this multiproject stuff into test, I backed off from the katharoseq support and put it back into a separate version.

Reply:
Just our luck! 😄 It sounds like it's doable down the road; I'll add an issue for it.